sdk/main/src/ED25519PrivateKey.cc

```diff
@@ -181,19 +174,21 @@ std::unique_ptr<ED25519PrivateKey> ED25519PrivateKey::derive(const uint32_t chil
     throw std::runtime_error("Key doesn't support derivation");
   }
 
-  if (!internal::DerivationPathUtils::isHardenedChildIndex(childIndex))
+  if (internal::DerivationPathUtils::isHardenedChildIndex(childIndex))
   {
-    throw std::runtime_error("Key supports only hardened derivation");
+    throw std::runtime_error("Child index should not be pre-hardened");
   }
 
+  const uint32_t hardenedIndex = internal::DerivationPathUtils::getHardenedIndex(childIndex);
```

Comment:
@alittley these changes here seemed like a bug, based on comments made [here](https://github.com/hashgraph/hedera-sdk-reference/issues/73#issue-1485459503) (specifically, when you said "this function automatically hardens index. Passing in a pre-hardened index should fail"). Let me know if this is incorrect.

Reply:
Not a bug per se - just an implementation I did pre-standard (TBH I like this way better, but the other SDKs already do the automatic hardening. but that's neither here nor there)

But to answer your question - yes, this will need to be changed to adhere to the new standard.

Reply:
Which is to say, the changes you made look correct